### PR TITLE
127 improve json template

### PR DESCRIPTION
**Pull request by Regorh**
Code has been added allowing users to download text present in JSON templates to their machine as a text file.  JSON template border and background coloring has been improved as well. A "copy" button could also be added at some point.

### CODE REVIEW

File: test.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<!--
+    author: You!
+-->
+<html>
```
Comment (thehabes) [inline test.html:5]: Thanks for committing this in so I could see how you were testing.  Just remember that this file should not be in the finalized PR for `main` and should only exist in your Draft PR.  Please remove this file.

File: js/deer-render.js [hunk 2]
```diff
@@ -104,18 +104,39 @@ RENDER.element = function (elem, obj) {
  * @param {Object} obj some json to be drawn as JSON
  * @param {Object} options additional properties to draw with the JSON
  */
+
 DEER.TEMPLATES.json = function (obj, options = {}) {
-    let indent = options.indent || 4
+    let indent = options.indent || 4;
```
Comment (thehabes) [inline js/deer-render.js:109]: Get rid of semicolons.

**Review by thehabes — CHANGES_REQUESTED** [review]
The button should be small and as out of the way as possible.  Instead of an actual button with text, consider using a symbol instead like https://icongr.am/feather/download.svg.  You can use a tooltip if you want it to have some text.

You did not consider how the button would affect text that is long enough to overflow.

image

You will have to think harder about positioning and make it so that the button is never in the way of text.

**Review by cubap — CHANGES_REQUESTED** [review]
Closing as unfinished though some useful parts are in here

**Pull request closed without merging**